src/charm.py

```diff
@@ -225,6 +256,55 @@ def _get_or_create_ca_certificate(self) -> Tuple[bytes, bytes]:
             self.app.add_secret(label=SELF_SIGNED_CA_SECRET_LABEL, content=content.to_dict())
         return content.certificate.encode(), content.private_key.encode()
 
+    def _get_or_create_admin_account(self) -> LoginSecret | None:
+        """Get the first admin user for the charm to use from secrets. Create one if it doesn't exist.
+
+        Returns:
+            Login details secret if they exist. None if the related account couldn't be created in GoCert.
+        """
+        try:
+            secret = self.model.get_secret(label=GOCERT_LOGIN_SECRET_LABEL)
+            secret_content = secret.get_content(refresh=True)
+            username = secret_content.get("username", "")
+            password = secret_content.get("password", "")
+            token = secret_content.get("token")
+            account = LoginSecret(username, password, token)
+        except ops.SecretNotFoundError:
+            username = _generate_username()
+            password = _generate_password()
+            account = LoginSecret(username, password, None)
+            self.app.add_secret(
+                label=GOCERT_LOGIN_SECRET_LABEL,
+                content=account.to_dict(),
+            )
+            logger.info("admin account details saved to secrets.")
+        if self.client.is_api_available() and not self.client.is_initialized():
+            response = self.client.create_first_user(username, password)
+            if not response:
+                logger.error("couldn't create the admin user.")
+                return None
+            logger.info("created the admin user in GoCert.")
+        return account
+
+
+def _generate_password() -> str:
+    """Generate a password for the GoCert Account."""
+    pw = []
+    pw.append(random.choice(string.ascii_lowercase))
+    pw.append(random.choice(string.ascii_uppercase))
+    pw.append(random.choice(string.digits))
+    pw.append(random.choice(string.punctuation))
+    for i in range(8):
+        pw.append(random.choice(string.ascii_letters + string.digits + string.punctuation))
+    random.shuffle(pw)
+    return "".join(pw)
+
+
+def _generate_username() -> str:
+    """Generate a username for the GoCert Account."""
+    suffix = [random.choice(string.ascii_uppercase) for i in range(4)]
+    return "charm-admin-" + "".join(suffix)
```

Comment:
I felt it would be a bit better for security to randomize a part of the username so there is absolutely no way to brute force it, since the code is open source and people can see what the username would be. I could change it if we don't want it.